File: Examples/test-suite/perl5/constover_runme.pl
use strict;
use warnings;
use Test::More tests => 8;
BEGIN { use_ok('constover') }
require_ok('constover');

is(constover::test("test"), "test", "test");
is(constover::test_pconst("test"), "test_pconst", "test_pconst");

my $f = constover::Foo->new();
is($f->test("test"), "test", "member-test");
is($f->test_pconst("test"), "test_pconst", "member-test_pconst");

is($f->test_constm("test"), "test_constmethod", "member-test_constm");
is($f->test_pconstm("test"), "test_pconstmethod", "member-test_pconstm");

File: Examples/test-suite/ocaml/constover_runme.ml
open Swig
open Constover

let test_str = "test" to string
let _ =
  assert ((_test (test_str) as string) = "test");
  assert ((_test_pconst (test_str) as string) = "test_pconst");
;;

let f = new_Foo '()
let _ =
  assert ((f -> "test" (test_str) as string) = "test");
  assert ((f -> "test_pconst" (test_str) as string) = "test_pconst");
  assert ((f -> "test_constm" (test_str) as string) = "test_constmethod");
  assert ((f -> "test_pconstm" (test_str) as string) = "test_pconstmethod");
;;
